market/src/callbacks.rs

```diff
@@ -5,15 +5,22 @@ use crate::storage::*;
 #[near_bindgen]
 impl Market {
     #[private]
-    pub fn on_create_proposal_callback(&mut self) -> bool {
-        match env::promise_result(0) {
-            PromiseResult::Successful(_res) => {
-                //let proposal_id: u64 = near_sdk::serde_json::from_slice(&res).unwrap();
-                //self.market_options.push(proposal_id);
+    pub fn on_create_proposal_callback(&mut self) {
+        let market_options_len = self.market.options.len() as u64;
```

Comment:
This suggests that you are storing the entire `market_options` strings in the contract? This could make storage expensive, no? This is why I proposed storing the `proposal_id` only. 

What we could do though, is store the `length` itself to perform this check: `self.market.options_length`

cc @susguzman

Reply:
I think that is better if we store the strings in the cotract. It is a more reliable way so that all the information is within the contract.